### DISCUSSION TITLE
umi ssr 中 getInitialProps 请求数据为什么会暴露在浏览器 network 中

### DISCUSSION THREAD

**Discussion by user-xxy** [umijs/umi#6179] [Q&A]
umi 项目，开启ssr，在getInitialProps中请求数据，网页源代码确实是服务端渲染后的，但是为什么在network中可以查看到请求接口，我曾经在nextJs中尝试ssr，在network中是看不见请求接口的。

代码片段：
![image](https://user-images.githubusercontent.com/55080157/109251923-4f094f80-7827-11eb-9372-57970c52d639.png)

浏览器network：
![image](https://user-images.githubusercontent.com/55080157/109252017-7cee9400-7827-11eb-8189-135943480b57.png)

不知道是我配置问题还是什么，求解答

**Comment by guanyuan** [comment]
出错，然后自动降级为客户端渲染了？

**Comment by joyz0** [comment]
@tuyeye 朋友你有没有遇到过我这些问题，[discussions/6633](https://github.com/umijs/umi/discussions/6633)，第一次弄ssr不太会。。。

**Comment by robinv8** [comment]
因为 getInitialProps 方法 会在客户端和服务端都会被触发，如果不想再客户端请求接口，可以
```
Page.getInitialProps = (ctx) => {
    const { isServer } = ctx;
    if (!isServer) {
       return;
    }
    ……
}
```
可以来 segmentfault [关注我](https://segmentfault.com/u/robin_ren) ，有问题随时问我
